apps/web/app/(app)/environments/[environmentId]/settings/api-keys/actions.ts

```diff
@@ -1,11 +1,30 @@
 "use server";
 
-import { deleteApiKey, createApiKey } from "@formbricks/lib/services/apiKey";
+import { authOptions } from "@/app/api/auth/[...nextauth]/authOptions";
+import { hasUserEnvironmentAccess } from "@/lib/api/apiHelper";
+import { deleteApiKey, createApiKey, canUserAccessApiKey } from "@formbricks/lib/services/apiKey";
 import { TApiKeyCreateInput } from "@formbricks/types/v1/apiKeys";
+import { Session, getServerSession } from "next-auth";
 
 export async function deleteApiKeyAction(id: string) {
-  return await deleteApiKey(id);
+  const session: Session | null = await getServerSession(authOptions);
+  const authorized = await canUserAccessApiKey(session, id);
+
+  if (authorized) {
+    return await deleteApiKey(id);
+  } else {
+    throw new Error("You are not authorized to perform this action.");
+  }
 }
 export async function createApiKeyAction(environmentId: string, apiKeyData: TApiKeyCreateInput) {
-  return await createApiKey(environmentId, apiKeyData);
+  const session = await getServerSession(authOptions);
+  if (!session) throw new Error("You are not authorized to perform this action.");
+
+  const authorized = await hasUserEnvironmentAccess(session.user, environmentId);
```

Comment:
If we now use `session` for the new methods instead of `session.user` we should also change it in `hasUserEnvironmentAccess` to avoid confusion

Reply:
Thought about this and decided to use **user** across the methods because we are validating the user itself hence that only should be passed rather than the session object!